### PR TITLE
az aro update: startvms, validate api servers, and configure api/ingress certs

### PR DESCRIPTION
**Pull request by ross-bryan**
### Which issue this PR addresses:

Allows customers to self-service certificate updates instead of relying on ARO SRE to patch the cluster.

### CODE REVIEW

File: pkg/cluster/install.go [hunk 1]
```diff
@@ -112,6 +112,10 @@ func (m *manager) Update(ctx context.Context) error {
 		steps.Action(m.clusterSPObjectID),
 		// credentials rotation flow steps
 		steps.Action(m.createOrUpdateClusterServicePrincipalRBAC),
+		steps.Action(m.startVMs),
```
Comment (m1kola) [inline pkg/cluster/install.go:115]: We still have `startVMs` step in `AdminUpdate`, but as far as I'm aware that is the current policy (decision by BU) is not to start VMs which were powered off.

Just something to check: Are we ok to have it in `Update`?

File: pkg/cluster/install.go [hunk 2]
```diff
@@ -112,6 +112,10 @@ func (m *manager) Update(ctx context.Context) error {
...
+		steps.Condition(m.apiServersReady, 30*time.Minute, true),
+		steps.Action(m.configureAPIServerCertificate),
+		steps.Action(m.configureIngressCertificate),
 		steps.Action(m.createOrUpdateDenyAssignment),
```
Comment (bennerv) [inline pkg/cluster/install.go:119]: Can we move `createOrUpdateDenyAssignment` above the kubernetes things?  it's strictly azure and is more likely to succeed

**Review by bennerv — CHANGES_REQUESTED** [review]

File: pkg/cluster/install.go (hunk 1 above)
Comment (ross-bryan) [reply pkg/cluster/install.go:115]: I think there is a difference between SRE turning them on, and cx running the `az aro update` command. Jerome indicated in a slack thread he was ok with update performing the startVMs check.

**Review by rogbas — DISMISSED** [review]
LGTM

**Review by cadenmarchese — APPROVED** [review]
Thanks, looks good to me. 💯

**Review by ross-bryan — COMMENTED** [review]
move denyassignment step before k8s steps

**Comment by ross-bryan** [comment]
lgtm

**Review by bennerv — APPROVED** [review]
lgtm - ship it

**Pull request merged**